# Target capture data

Much of the information in this chapter is drawn from the HybPiper tutorial, which can be found at https://github.com/mossmatters/HybPiper/wiki/Tutorial

First make a folder in your home directory to hold your outputs for this component of the project.

We have pre-installed the program HybPiper, which assembles target sequences from high-throughput DNA sequencing reads. 
To access HybPiper run

```
conda activate /mnt/homes4celsrs/shared/envs/hybpiper
```

In this case, our target sequences are the Angiosperm 353 standard targets. 
Target capture is a standard technique for sequencing many known loci simultaneously using a library of sequences. 
We have provided you with low-coverage sequence data from using these target sequences as probes (your fastq files).
HybPiper first assigns each read to a particular target using alignment software (BWA).
These reads are then assembled using a standard genome assembler (Spades).
HybPiper outputs a fasta file of the (in frame) CDS portion of each sample for each target region.

The basic command for HybPiper is as follows:

```
hybpiper assemble -t_dna targets.fasta -r species1_R*_test.fastq --prefix species1 --bwa --cpu 2
```

You should run this command, changing the file starting with "species1" to the name of one of your files, which you can find by listing the files in the folder of your data. Note the * indicates that we will use both pairs of the sequencing read files. Also change the prefix to something relevant for this species. Note that we are sharing a server so we are specifying 2 cpu's per group - if you are running on your own machine you can omit this flag and HybPiper will use all available resources.

To automatically run all samples consecutively you will need to loop through them.
One approach is to make a list of all the names of the species in a file and then read that and use them one at a time.

First we need to make the list. We want to automate this process because in some cases you could have a lot of samples. Additionally, copying and pasting names into a list is prone to error. Plus once you automate this process you could repeat it easily and quickly for other datasets.
For starters make a list of just the R1 files (you don't want R2 files because they have the same name so you would have duplicates) and output this to a file (name it namelist.txt).
Really, we don't want such long names, plus this includes R1 alone and we want to be able to list both files together.
One approach is to notice a pattern in filenames: the files start with a sample identifier followed by an underscore followed by another identifier.
Thus, we can "cut off" all of the name after the second underscore and still have a unique name for our sample.
We use the "cut" command to make our data into columns using _ (specify a delimiter with -d), then we select the first two of these (use the field flag -f).

```
ls *R1* | cut -d '_' -f 1,2 >namelist.txt 
```

You can check this worked by viewing the file and counting the number of lines.

Now we can run HybPiper on all of our data sequentially.
In the following example, the species names are entered in the namelist.txt file.
The loop will iterate through them one at a time and run HybPiper.

```
while read name
do 
    hybpiper assemble -t_dna ../../Campanulaceae_353.fasta -r ${name}*fq.gz --prefix $name --bwa --cpu 2
done < namelist.txt
```

At this point, each sequence is in its own folder.
We can use HybPiper to fetch the sequences recovered for the same gene from multiple samples and generate a file for each gene. Use the following command, 

```
hybpiper retrieve_sequences dna -t_dna ../../Campanulaceae_353.fasta --sample_names namelist.txt
```

Note that HybPiper has additional features we will not use in this workshop due to a lack of time.


## Visual inspection of data (eg, alignment, BLAST)


## Align data

We will use MAFFT to align our data.
Due to the nature of this server different programs are in different "environments" so you'll need to deactivate the hybpiper environment and activate one for mafft.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/mafft
```

Now MAFFT is available to run.
Note that we are looping through each of the data files (one per gene) generated by HybPiper and outputting an alignment.

```{bash, echo=FALSE, eval=FALSE}
for F in *FNA; do
mafft --auto --thread 1 $F > ${F}.fasta
done
```

