# Target capture data

Much of the information in this chapter is drawn from the HybPiper tutorial, which can be found at https://github.com/mossmatters/HybPiper/wiki/Tutorial

```{bash, echo=FALSE, eval=FALSE}
also see tutorials at
https://hackmd.io/@mossmatters?utm_source=preview-mode&utm_medium=rec
https://hackmd.io/@mossmatters/SJLQRQTGY
https://hackmd.io/@mossmatters/HkPM7pwEK
```

First make a folder in your home directory to hold your outputs for this component of the project.

```{bash, echo=FALSE, eval=FALSE}
sudo conda create -p /mnt/homes4celsrs/shared/envs/hybpiper -c chrisjackson-pellicle hybpiper
```

We have pre-installed the program HybPiper, which assembles target sequences from short high-throughput DNA sequencing reads. 
By assembling each target sequence for each species we will be able to produce a dataset that allows us to understand the evolutionary history of the species.

To access HybPiper run.
This activates a conda environment (don't worry about this!).

```
conda activate /mnt/homes4celsrs/shared/envs/hybpiper
```

In this case, our target sequences are the Angiosperm 353 standard targets. 
Target capture is a standard technique for sequencing many known loci simultaneously using a library of sequences. 
We have provided you with low-coverage sequence data from using these target sequences as probes (your fastq files).
HybPiper first assigns each read to a particular target using alignment software (BWA).
These reads are then assembled using a standard genome assembler (Spades).
HybPiper outputs a fasta file of the (in frame) CDS portion of each sample for each target region.

The basic command for HybPiper is as follows:

```
hybpiper assemble -t_dna targets.fasta -r species1_R*_test.fastq --prefix species1 --bwa --cpu 3
```

Notice the commands and flags.
Move to the results directory in your workshop directory to run the following commands.
You should run this command, with the following changes:

* The argument for t_dna is the fasta file in the shared workshop folder - you will need to indicate a full relative path not just the file name
* The argument for the the r flag is the filename(s) including paths for the fastq files of interest. Note the * indicates that we will use both pairs of the sequencing read files. 
* The prefix should be indicative of the species name
* Note that we are sharing a server so we are specifying 3 cpu's per group - if you are running on your own machine you can omit this flag and HybPiper will use all available resources. With limited shared resources this run may take a few minutes.

To view your results, list the contents of the folder and folders inside this.
You should see a folder for each gene.
Within each folder you can see several fasta files, which you can view. 

To automatically run all samples consecutively you will need to loop through them.
Go back and look at the prior example of loops.
One approach is to make a list of all the names of the species in a file and then read that and use them one at a time.

First we need to make the list. We want to automate this process because in some cases you could have a lot of samples. Additionally, copying and pasting names into a list is prone to error. 
Additionally, once you automate this process you could repeat it easily and quickly for other datasets.
For starters make a list of just the R1 files (you don't want R2 files because they have the same name so you would have duplicates). 
First cd back to your data folder.
Output your list to a file in your results folder named `namelist.txt` by "redirecting" the output using `>`.

```
ls *R1* > ~/evolution_workshop/results/namelist.txt 
```

Check that this worked using `cat` or by viewing the file and counting the number of lines..
We will need to adjust this output do remove the R1 label so that we can use wildcards to list both files together.
One approach is to notice a pattern in filenames: the files start with a sample identifier followed by an underscore followed by another identifier.
Thus, we can "cut off" all of the name after the second underscore and still have a unique name for our sample.
We use the "cut" command to make our data into columns using _ (specify a delimiter with -d), then we select the first two of these (use the field flag -f).

```
ls *R1* | cut -d '_' -f 1,2 > ~/evolution_workshop/results/namelist.txt 
```

Check this worked.
Now we can run HybPiper on all of our data sequentially.
In the following example, the species names are entered in the namelist.txt file.
The loop will iterate through them one at a time and run HybPiper.

```
while read name
do 
    [insert command here]
done < [path to]/namelist.txt
```

To obtain some information about the results of this process use the `hybpiper stats` command as follows.

```
hybpiper stats -t_dna [target file] gene [path to]/namelist.txt
```

From https://hackmd.io/@mossmatters/HkPM7pwEK#Getting-HybPiper-Stats

*hybpiper stats will generate two files, seq_lengths.tsv and hybpiper_stats.tsv

The first line of seq_lengths.tsv has the names of each gene. The second line has the length of the target gene, averaged over each “source” for that gene. The rest of the lines are the length of the sequence recovered by HybPiper for each gene. If there was no sequence for a gene, a 0 is entered.*

At this point, each sequence is in its own folder.
We can use HybPiper to fetch the sequences recovered for the same gene from multiple samples and generate a file for each gene. Use the following command, 

```
hybpiper retrieve_sequences dna -t_dna [target file]  --sample_names [path to]/namelist.txt
```

Note that HybPiper has additional features we will not use in this workshop due to a lack of time.


## Visual inspection of data

To check that your genes assembled correctly and are likely the correct species you may use BLAST, which is found at https://blast.ncbi.nlm.nih.gov/Blast.cgi

* Select Nucleotide BLAST
* Paste one of the contigs from one of the genes from one of the species
* Select Somewhat similar sequences (blastn)
* BLAST
* Examine the species and the match
