# Results

## Alignment

In order to use these data to build our phylogenies, we need to align each gene.
This allows our tree-building software to compare species using nucleotides that we believe share ancestry. 
We will use the software MAFFT.

```
for con in contig_list
do
mafft --auto $con > ${con}.aligned
done
```

## Concatenated data analysis

There are multiple ways to analyze data.
The first is to concatenate everything.

```
python3 AMAS.py concat -f fasta -d dna --out-format fasta --part-format raxml -i $infiles -t concatenated.fasta -p partitions.txt

```

### Building trees with IQTree

We will build our first tree using our complete concatenated dataset.
The program IQtree infers phylogenetic trees by maximum likelihood.
This approach starts with a tree and calculates the likelihood of the data on the tree (i.e. calculating the probability of each site fitting the tree given a model of substitution and multiplying them together).
We use the General Time Reversible (GTR) model to allow sites to change back and forth among different bases with particular probabilities.
We also allow a Gamma (G) distribution of rates of substitution across sites.
We allow partitioning of the data by gene so that different genes can evolve according to different models.

```
iqtree2 -nt 2 -s concatenated.fasta -spp partitions.txt -pre iqtree_tree -n 0 -m GTR+G
```

### Viewing trees

Using RStudio...

### svdq

```
echo "
log file=${logFile};
tonexus format=RelPHYLIP fromfile=concatenated.fasta tofile=concatenated.svdq.nex interleaved=yes replace=yes;
exe concatenated.svdq.nex;
svdq bootstrap nreps=1000 nthreads=2;
saveTrees file=concatenated.svdq;
quit;
" | paup
```

### MrBayes

```

```

## Species tree analyses

An alternative is to use a species tree approach.

### ASTRAL??

```
#gene trees from iqtree

#astral
```

## Support for relationships

### Bootstrap

### SCF / GCF

```
iqtree2 -t tree.tre -s concatenated.fasta --scf 100 --prefix SCF -nt 2
```


## What does your tree tell you?

Compare your trees

