# Results

### Alignment

In order to use these data to build our phylogenies, we need to align each gene.
We will use MAFFT ...

```
for con in contig_list
do
mafft --auto $con > ${con}.aligned
done
```

### Concatenating data

There are multiple ways to analyze data.
The first is to concatenate everything.

```
python3 AMAS.py concat -f fasta -d dna --out-format fasta --part-format raxml -i $infiles -t concatenated.fasta -p partitions.txt

```

#### Building trees with IQTree

```
iqtree2 -nt 2 -s concatenated.fasta -spp partitions.txt -z tree.tre -pre calcLnL -n 0 -m GTR+G -wsl

```

#### svdq??

```
echo "
log file=${logFile};
tonexus format=RelPHYLIP fromfile=concatenated.fasta tofile=concatenated.svdq.nex interleaved=yes replace=yes;
exe concatenated.svdq.nex;
svdq bootstrap nreps=1000 nthreads=2;
saveTrees file=concatenated.svdq;
quit;
" | paup
```

### Species trees

An alternative is to use a species tree approach.

#### ASTRAL??

```

```

### Support for relationships

#### Bootstrap

#### SCF / GCF

```
iqtree2 -t tree.tre -s concatenated.fasta --scf 100 --prefix SCF -nt 2
```






## What does your tree tell you?

You will need to compare your tree to what's in the literature.
Go back to the first readings you did and continue to explore.
Are your results novel (e.g. have you determined relationships among species that no one had investigated)?
Do your results confirm or refute prior work?
