# Results

## Alignment

In order to use these data to build our phylogenies, we need to align each gene.
This allows our tree-building software to compare species using nucleotides that we believe share ancestry. 
We will use the software MAFFT.
Due to the nature of this server different programs are in different "environments" so you'll need to deactivate the hybpiper environment and activate one for mafft.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/mafft
```

Now MAFFT is available to run.
Note that we are looping through each of the data files (one per gene) generated by HybPiper and outputting an alignment.
The basic mafft command looks like the following, assuming you replace FILE with a particular file.

```
mafft --auto --thread 1 FILE > FILE.fasta

for f in *fasta; do
cut -f 1 -d ' ' $f > ${f}.fa
done
```

However, you should be able to loop through all of your files output by HybPiper so that you can align all of them sequentially in one loop command.


```{bash, echo=FALSE, eval=FALSE}
for F in *FNA; do
mafft --auto --thread 1 $F > ${F}.fasta
done
```

## Concatenated data analysis

There are multiple ways to analyze data.
The first is to concatenate everything.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/amas

python3 /mnt/homes4celsrs/shared/envs/amas/bin/AMAS.py concat -f fasta \
-d dna --out-format fasta --part-format raxml -i *FNA.fasta.fa \
-t concatenated.fasta -p partitions.txt

```
Note that if some data was not found for some genes you may need to delete files to get AMAS to concatenate your data.


### Building trees with IQTree

We will build our first tree using our complete concatenated dataset.
The program IQtree infers phylogenetic trees by maximum likelihood.
This approach starts with a tree and calculates the likelihood of the data on the tree (i.e. calculating the probability of each site fitting the tree given a model of substitution and multiplying them together).
We use the General Time Reversible (GTR) model to allow sites to change back and forth among different bases with particular probabilities.
We also allow a Gamma (G) distribution of rates of substitution across sites.
We allow partitioning of the data by gene so that different genes can evolve according to different models.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/iqtree

iqtree2 -nt 2 -s concatenated.fasta -spp partitions.txt -pre iqtree_tree -n 0 -m GTR+G
```

Tree for all data using output of individual hybpiper runs

### Viewing trees

* Make new RScript

```{r, eval=FALSE}
library(tidyverse)
library(ape)

tax1 <- read.tree("AndesWorkshop2023/TaxonSet1/353reads/iqtree_tree.treefile")
plot(tax1)
```

### svdq

```
echo "
log file=${logFile};
tonexus format=RelPHYLIP fromfile=concatenated.fasta tofile=concatenated.svdq.nex \
interleaved=yes replace=yes;
exe concatenated.svdq.nex;
svdq bootstrap nreps=1000 nthreads=2;
saveTrees file=concatenated.svdq;
quit;
" | paup
```

### MrBayes

```

```

## Species tree analyses

An alternative is to use a species tree approach.

### ASTRAL

```
#gene trees from iqtree

#astral
```

## Support for relationships

### Bootstrap

### SCF / GCF

```
iqtree2 -t tree.tre -s concatenated.fasta --scf 100 --prefix SCF -nt 2
```


## What does your tree tell you?

Compare your trees

