# Results

## Alignment

In order to use these data to build our phylogenies, we need to align each gene.
This allows our tree-building software to compare species using nucleotides that we believe share ancestry. 
We will use the software MAFFT.
Due to the nature of this server different programs are in different "environments" so you'll need to deactivate the hybpiper environment and activate one for mafft.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/mafft
```

Now MAFFT is available to run.
However, before we align our data let's take a look at the output.
Do you remember how to print out particular lines of a file?
Try printing all of the lines that start with '>' in a particular FNA file. **Make sure to use '>' including the single quotes when searching.**
You should notice that HybPiper has added some information to these lines so they include more than just the name of the sample. Because these additions differ across genes subsequent analyses may treat them as separate samples.

Before we do the alignment we need to remove this extra information.
Again, we can think about the pattern we are looking for: we want to go through each FNA file (this should suggest using a loop) and keep the first "word" in each line (think of a word as a set of characters not separated by spaces).
We use the `cut` command to divide each line of our file into columns.
We use the `-f` flag to indicate the field (column) we want to keep, and the `-d` flag to indicate how to separate columns (i.e. with a space: ' ').

An example cut command for one file could look like the following:

```
cut -f 1 -d ' ' 7577.FNA > 7577.FNA.fa
```

Now use a loop as before to repeat this command for all FNA files.

```{bash, echo=FALSE, eval=FALSE}
for f in *FNA; do
cut -f 1 -d ' ' $f > ${f}.fa
done
```

Now we can loop through each of these files and output an alignment.
The basic mafft command looks like the following, assuming you replace FILE with a particular file.

```
mafft --auto --thread 1 7577.FNA.fa > 7577.FNA.fa.fasta
```

Remember you have 350 genes (i.e. 350 fasta files) so you want to run these alignments in a loop.
As before, write a loop to go through each FNA.fa file and output an alignment.


```{bash, echo=FALSE, eval=FALSE}
for F in *FNA.fa; do
mafft --auto --thread 1 $F > ${F}.fasta
done
```

## View in R

The best way to view an alignment is in a specialized program, but because we have R
easily available we'll view an example here.

Make a new script to run this alignment view.
You should first load the `ape` and `ggmsa` library. (This uses the gg as in ggplot and msa, which stands for multiple sequence alignment.)

* Read in your fasta file with ape's `read.FASTA`.
* Use the ggmsa command as follows, substituting the particular variable that contains your fasta file. Allow a little while for this to run. You may need to click the Zoom button above the plot to see it more clearly.

```
ggmsa(seqs_7577, start = 20, end = 120, char_width = 0.5, seq_name = T, color = "Chemistry_NT") +
  geom_msaBar()
```

Now repeat this process for the same gene with the unaligned data. Can you see the difference?

```{bash, echo=FALSE, eval=FALSE}
library(ggmsa)  #use R 4.1.2
library(ape)

#view aligned
tax1_seqs_4471 <- ("AndesWorkshop2023/TaxonSet1/353reads/4471.FNA.fasta.fa")
ggmsa(tax1_seqs_4471, start = 20, end = 120, char_width = 0.5, seq_name = T, color = "Chemistry_NT") + 
  geom_msaBar() #this is slow - wait!

#view unaligned
tax1_seqs_4471u <- ("AndesWorkshop2023/TaxonSet1/353reads/4471.FNA")
ggmsa(tax1_seqs_4471u, start = 20, end = 120, char_width = 0.5, seq_name = T, color = "Chemistry_NT") + 
  geom_msaBar() #this is slow - wait!
```

## Trimal
https://hackmd.io/@mossmatters/Sy6md0prY

## Concatenated data analysis

There are multiple ways to analyze data.
The first is to concatenate everything.
Make sure you are in your Terminal.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/amas

python3 /mnt/homes4celsrs/shared/envs/amas/bin/AMAS.py concat -f fasta \
-d dna --out-format fasta --part-format raxml -i *FNA.fa.fasta \
-t concatenated.fasta -p partitions.txt

```
Note that if some data was not found for some genes you may need to delete files to get AMAS to concatenate your data.


### Building trees with IQTree

We will build our first tree using our complete concatenated dataset.
The program IQtree infers phylogenetic trees by maximum likelihood.
This approach starts with a tree and calculates the likelihood of the data on the tree (i.e. calculating the probability of each site fitting the tree given a model of substitution and multiplying them together).
We use the General Time Reversible (GTR) model to allow sites to change back and forth among different bases with particular probabilities.
We also allow a Gamma (G) distribution of rates of substitution across sites.
We allow partitioning of the data by gene so that different genes can evolve according to different models.

```
conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/iqtree

iqtree2 -nt 2 -s concatenated.fasta -spp partitions.txt -pre iqtree_tree -B 1000 -m GTR+G
```

Tree for all data using output of individual hybpiper runs

### Viewing trees

* Make a new RScript for to view your tree in R.
* Load the libraries tidyverse and ape.
* Use the `read.tree` command and provide the path to the tree as the argument
* Use the plot command providing the tree variable as the argument, then add `, show.node.label = TRUE`

```{r, eval=FALSE, echo=FALSE}

tax1 <- read.tree("AndesWorkshop2023/TaxonSet1/353reads/iqtree_tree.treefile")
plot(tax1, show.node.label = TRUE)
```

### svdq

```
echo "
log file=${logFile};
tonexus format=RelPHYLIP fromfile=concatenated.fasta tofile=concatenated.svdq.nex \
interleaved=yes replace=yes;
exe concatenated.svdq.nex;
svdq bootstrap nreps=1000 nthreads=2;
saveTrees file=concatenated.svdq;
quit;
" | paup
```

* Load and view this tree in R.

### MrBayes

```

```

## Species tree analyses

An alternative is to use a species tree approach.

### ASTRAL

```
#gene trees from iqtree

#astral
```

## Support for relationships

### Bootstrap

### SCF / GCF

```
iqtree2 -t tree.tre -s concatenated.fasta --scf 100 --prefix SCF -nt 2
```


## What does your tree tell you?

Compare your trees

