<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 9 Results | Andes Phylogenomics Workshop</title>
  <meta name="description" content="Step-by-step guide to a basic project in phylogenomics." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 9 Results | Andes Phylogenomics Workshop" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="Step-by-step guide to a basic project in phylogenomics." />
  <meta name="github-repo" content="rachelss/Andes_workshop_codebook" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 9 Results | Andes Phylogenomics Workshop" />
  
  <meta name="twitter:description" content="Step-by-step guide to a basic project in phylogenomics." />
  

<meta name="author" content="Rachel Schwartz" />


<meta name="date" content="2023-06-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="variant-calling.html"/>
<link rel="next" href="sharing-your-results.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Andes Phylogenomics Workshop</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Getting Started</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction</a>
<ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#case-studies"><i class="fa fa-check"></i><b>2.2</b> Case Studies</a></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#course-objectives"><i class="fa fa-check"></i><b>2.3</b> Course Objectives</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-and-analysis.html"><a href="data-and-analysis.html"><i class="fa fa-check"></i><b>3</b> Data and analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="data-and-analysis.html"><a href="data-and-analysis.html#genomic-data"><i class="fa fa-check"></i><b>3.1</b> Genomic data</a></li>
<li class="chapter" data-level="3.2" data-path="data-and-analysis.html"><a href="data-and-analysis.html#phylogenies"><i class="fa fa-check"></i><b>3.2</b> Phylogenies</a></li>
<li class="chapter" data-level="3.3" data-path="data-and-analysis.html"><a href="data-and-analysis.html#comparative-analyses"><i class="fa fa-check"></i><b>3.3</b> Comparative analyses</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="computational-skills.html"><a href="computational-skills.html"><i class="fa fa-check"></i><b>4</b> Computational skills</a>
<ul>
<li class="chapter" data-level="4.1" data-path="computational-skills.html"><a href="computational-skills.html#log-in-to-the-server"><i class="fa fa-check"></i><b>4.1</b> Log in to the server</a></li>
<li class="chapter" data-level="4.2" data-path="computational-skills.html"><a href="computational-skills.html#basic-navigation"><i class="fa fa-check"></i><b>4.2</b> Basic navigation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="getting-your-data.html"><a href="getting-your-data.html"><i class="fa fa-check"></i><b>5</b> Getting your data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="getting-your-data.html"><a href="getting-your-data.html#repeating-analyses"><i class="fa fa-check"></i><b>5.1</b> Repeating analyses</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="target-capture-data.html"><a href="target-capture-data.html"><i class="fa fa-check"></i><b>6</b> Target capture data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="target-capture-data.html"><a href="target-capture-data.html#visual-inspection-of-data-eg-alignment-blast"><i class="fa fa-check"></i><b>6.1</b> Visual inspection of data (eg, alignment, BLAST)</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="plastome-variation.html"><a href="plastome-variation.html"><i class="fa fa-check"></i><b>7</b> Plastome variation</a></li>
<li class="chapter" data-level="8" data-path="variant-calling.html"><a href="variant-calling.html"><i class="fa fa-check"></i><b>8</b> Variant Calling</a></li>
<li class="chapter" data-level="9" data-path="results.html"><a href="results.html"><i class="fa fa-check"></i><b>9</b> Results</a>
<ul>
<li class="chapter" data-level="9.1" data-path="results.html"><a href="results.html#alignment"><i class="fa fa-check"></i><b>9.1</b> Alignment</a></li>
<li class="chapter" data-level="9.2" data-path="results.html"><a href="results.html#view-in-r"><i class="fa fa-check"></i><b>9.2</b> View in R</a></li>
<li class="chapter" data-level="9.3" data-path="results.html"><a href="results.html#trimal"><i class="fa fa-check"></i><b>9.3</b> Trimal</a></li>
<li class="chapter" data-level="9.4" data-path="results.html"><a href="results.html#concatenated-data-analysis"><i class="fa fa-check"></i><b>9.4</b> Concatenated data analysis</a>
<ul>
<li class="chapter" data-level="9.4.1" data-path="results.html"><a href="results.html#building-trees-with-iqtree"><i class="fa fa-check"></i><b>9.4.1</b> Building trees with IQTree</a></li>
<li class="chapter" data-level="9.4.2" data-path="results.html"><a href="results.html#viewing-trees"><i class="fa fa-check"></i><b>9.4.2</b> Viewing trees</a></li>
<li class="chapter" data-level="9.4.3" data-path="results.html"><a href="results.html#svdq"><i class="fa fa-check"></i><b>9.4.3</b> svdq</a></li>
<li class="chapter" data-level="9.4.4" data-path="results.html"><a href="results.html#mrbayes"><i class="fa fa-check"></i><b>9.4.4</b> MrBayes</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="results.html"><a href="results.html#species-tree-analyses"><i class="fa fa-check"></i><b>9.5</b> Species tree analyses</a>
<ul>
<li class="chapter" data-level="9.5.1" data-path="results.html"><a href="results.html#astral"><i class="fa fa-check"></i><b>9.5.1</b> ASTRAL</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="results.html"><a href="results.html#support-for-relationships"><i class="fa fa-check"></i><b>9.6</b> Support for relationships</a>
<ul>
<li class="chapter" data-level="9.6.1" data-path="results.html"><a href="results.html#bootstrap"><i class="fa fa-check"></i><b>9.6.1</b> Bootstrap</a></li>
<li class="chapter" data-level="9.6.2" data-path="results.html"><a href="results.html#scf-gcf"><i class="fa fa-check"></i><b>9.6.2</b> SCF / GCF</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="results.html"><a href="results.html#what-does-your-tree-tell-you"><i class="fa fa-check"></i><b>9.7</b> What does your tree tell you?</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="sharing-your-results.html"><a href="sharing-your-results.html"><i class="fa fa-check"></i><b>10</b> Sharing your results</a>
<ul>
<li class="chapter" data-level="10.1" data-path="sharing-your-results.html"><a href="sharing-your-results.html#presentation"><i class="fa fa-check"></i><b>10.1</b> Presentation</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Andes Phylogenomics Workshop</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="results" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">Chapter 9</span> Results<a href="results.html#results" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="alignment" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Alignment<a href="results.html#alignment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In order to use these data to build our phylogenies, we need to align each gene.
This allows our tree-building software to compare species using nucleotides that we believe share ancestry.
We will use the software MAFFT.
Due to the nature of this server different programs are in different “environments” so you’ll need to deactivate the hybpiper environment and activate one for mafft.</p>
<pre><code>conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/mafft</code></pre>
<p>Now MAFFT is available to run.
Note that we are looping through each of the data files (one per gene) generated by HybPiper and outputting an alignment.
The basic mafft command looks like the following, assuming you replace FILE with a particular file.</p>
<pre><code>mafft --auto --thread 1 FILE &gt; FILE.fasta

for f in *fasta; do
cut -f 1 -d &#39; &#39; $f &gt; ${f}.fa
done</code></pre>
<p>However, you should be able to loop through all of your files output by HybPiper so that you can align all of them sequentially in one loop command.</p>
</div>
<div id="view-in-r" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> View in R<a href="results.html#view-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<pre><code>library(ggmsa)  #use R 4.1.2

#view aligned
tax1_seqs_4471 &lt;- (&quot;AndesWorkshop2023/TaxonSet1/353reads/4471.FNA.fasta.fa&quot;)
ggmsa(tax1_seqs_4471, start = 20, end = 120, char_width = 0.5, seq_name = T, color = &quot;Chemistry_NT&quot;) + 
  geom_msaBar() #this is slow - wait!

#view unaligned
tax1_seqs_4471u &lt;- (&quot;AndesWorkshop2023/TaxonSet1/353reads/4471.FNA&quot;)
ggmsa(tax1_seqs_4471u, start = 20, end = 120, char_width = 0.5, seq_name = T, color = &quot;Chemistry_NT&quot;) + 
  geom_msaBar() #this is slow - wait!</code></pre>
</div>
<div id="trimal" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Trimal<a href="results.html#trimal" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://hackmd.io/@mossmatters/Sy6md0prY" class="uri">https://hackmd.io/@mossmatters/Sy6md0prY</a></p>
</div>
<div id="concatenated-data-analysis" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Concatenated data analysis<a href="results.html#concatenated-data-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There are multiple ways to analyze data.
The first is to concatenate everything.</p>
<pre><code>conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/amas

python3 /mnt/homes4celsrs/shared/envs/amas/bin/AMAS.py concat -f fasta \
-d dna --out-format fasta --part-format raxml -i *FNA.fasta.fa \
-t concatenated.fasta -p partitions.txt
</code></pre>
<p>Note that if some data was not found for some genes you may need to delete files to get AMAS to concatenate your data.</p>
<div id="building-trees-with-iqtree" class="section level3 hasAnchor" number="9.4.1">
<h3><span class="header-section-number">9.4.1</span> Building trees with IQTree<a href="results.html#building-trees-with-iqtree" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will build our first tree using our complete concatenated dataset.
The program IQtree infers phylogenetic trees by maximum likelihood.
This approach starts with a tree and calculates the likelihood of the data on the tree (i.e. calculating the probability of each site fitting the tree given a model of substitution and multiplying them together).
We use the General Time Reversible (GTR) model to allow sites to change back and forth among different bases with particular probabilities.
We also allow a Gamma (G) distribution of rates of substitution across sites.
We allow partitioning of the data by gene so that different genes can evolve according to different models.</p>
<pre><code>conda deactivate
conda activate /mnt/homes4celsrs/shared/envs/iqtree

iqtree2 -nt 2 -s concatenated.fasta -spp partitions.txt -pre iqtree_tree -B 1000 -m GTR+G</code></pre>
<p>Tree for all data using output of individual hybpiper runs</p>
</div>
<div id="viewing-trees" class="section level3 hasAnchor" number="9.4.2">
<h3><span class="header-section-number">9.4.2</span> Viewing trees<a href="results.html#viewing-trees" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>Make new RScript</li>
</ul>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="results.html#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb33-2"><a href="results.html#cb33-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span>
<span id="cb33-3"><a href="results.html#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="results.html#cb33-4" aria-hidden="true" tabindex="-1"></a>tax1 <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(<span class="st">&quot;AndesWorkshop2023/TaxonSet1/353reads/iqtree_tree.treefile&quot;</span>)</span>
<span id="cb33-5"><a href="results.html#cb33-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tax1, <span class="at">show.node.label =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="svdq" class="section level3 hasAnchor" number="9.4.3">
<h3><span class="header-section-number">9.4.3</span> svdq<a href="results.html#svdq" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>echo &quot;
log file=${logFile};
tonexus format=RelPHYLIP fromfile=concatenated.fasta tofile=concatenated.svdq.nex \
interleaved=yes replace=yes;
exe concatenated.svdq.nex;
svdq bootstrap nreps=1000 nthreads=2;
saveTrees file=concatenated.svdq;
quit;
&quot; | paup</code></pre>
</div>
<div id="mrbayes" class="section level3 hasAnchor" number="9.4.4">
<h3><span class="header-section-number">9.4.4</span> MrBayes<a href="results.html#mrbayes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code></code></pre>
</div>
</div>
<div id="species-tree-analyses" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Species tree analyses<a href="results.html#species-tree-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>An alternative is to use a species tree approach.</p>
<div id="astral" class="section level3 hasAnchor" number="9.5.1">
<h3><span class="header-section-number">9.5.1</span> ASTRAL<a href="results.html#astral" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>#gene trees from iqtree

#astral</code></pre>
</div>
</div>
<div id="support-for-relationships" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Support for relationships<a href="results.html#support-for-relationships" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="bootstrap" class="section level3 hasAnchor" number="9.6.1">
<h3><span class="header-section-number">9.6.1</span> Bootstrap<a href="results.html#bootstrap" class="anchor-section" aria-label="Anchor link to header"></a></h3>
</div>
<div id="scf-gcf" class="section level3 hasAnchor" number="9.6.2">
<h3><span class="header-section-number">9.6.2</span> SCF / GCF<a href="results.html#scf-gcf" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<pre><code>iqtree2 -t tree.tre -s concatenated.fasta --scf 100 --prefix SCF -nt 2</code></pre>
</div>
</div>
<div id="what-does-your-tree-tell-you" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> What does your tree tell you?<a href="results.html#what-does-your-tree-tell-you" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Compare your trees</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="variant-calling.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sharing-your-results.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/08-results.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["bookdown-demo.pdf", "bookdown-demo.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
